repl/appengine/app/api_test.go

```diff
@@ -0,0 +1,147 @@
+package app
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func responseEqual(lhs, rhs *evaluateResponse) bool {
+	if lhs == nil {
+		if rhs == nil {
+			return true
+		}
+		return false
+	}
+
+	return reflect.DeepEqual(lhs.Responses, rhs.Responses)
```

Comment:
If you use getters for `GetResponses()` then you can conditionally return `nil` if the function target is `nil` and the whole body of the function can be `reflect.DeepEqual(lhs.GetResponses(), rhs.GetResponses())`